### ISSUE TITLE
stamping chart of type geo then table yields undefined chart type

### ISSUE THREAD

**Issue by jongeho1** [GoogleWebComponents/google-chart#72]
Hi there, 

Facing inconsistent behavior when using multiple ```google-chart``` with varying types that are stamped into the dom at different times. A good example is stamping one of type geo then of table.

```html

<link rel="import" href="bower_components/polymer/polymer.html">
<link rel="import" href="bower_components/google-chart/google-chart.html">

<dom-module id="the-el">
    <template>
        <button on-tap="_tTable">Stamp Table</button>
        <button on-tap="_tGeo">Stamp Geo</button>
        <button on-tap="_tBoth">Stamp Both</button>
        <template is="dom-if" if="{{showGeo}}" restamp>
            <h1>Stamping Geo</h1>
            <h2>Geo Chart</h2>
            <google-chart type="geo" data='[["Country","# of cows"],["BS",752348],["CD",306523],["AO",806287],["KY",169022],["IO",982],["BE",7240846],["CU",8951],["AW",237738],["CC",1],["AL",3170943],["KH",5039318],["BO",3677044],["BH",2528854],["CK",6496],["AI",20973],["CF",5877],["BN",1261279],["BZ",396333],["CY",1003070],["AU",18601713],["CO",25019570],["DZ",6032205],["CM",318071],["BA",1746715],["BD",6862277],["CR",4356245],["AG",248264],["TD",21008],["BG",3331495],["BJ",203453],["CZ",2748968],["AT",4009955],["KM",12892],["AS",31794],["CA",47546018],["BW",98467],["BB",613601],["CI",758435],["AR",23899962],["CL",17863532],["UN",4712275],["BF",166681],["BM",180303],["DK",3527352],["AZ",5370918],["CG",92149],["AD",96320],["CV",71066],["BR",65542588],["BY",2625345],["HR",3419193],["AM",915960],["CN",20931921],["AF",662182],["BI",30621],["BT",76527],["DJ",94734],["GR",4253871],["LA",3733866],["FR",27547672],["JP",38761999],["SV",2267621],["IS",321892],["GT",3862248],["GE",1306661],["KP",239],["FK",1181],["IE",6520811],["DM",113432],["VA",364],["GL",49078],["LV",781081],["GF",218059],["JO",4101559],["GQ",43056],["IN",37044982],["GN",543681],["DE",28552536],["KR",18184529],["FO",43597],["IL",8858476],["DO",6682727],["HN",2079527],["GD",123815],["LB",7304374],["PF",471257],["KZ",5851021],["ER",264],["ID",62250775],["GW",8619],["GH",1703842],["KW",6862643],["FJ",172913],["IT",23010367],["EC",6191501],["HK",10992110],["GP",509681],["LS",13607],["GA",230932],["KE",923662],["EE",1387922],["IR",14869013],["GY",298209],["GI",82716],["KG",946043],["FI",1952908],["JM",2698134],["EG",8721635],["HU",3543622],["GU",489182],["GM",66002],["KI",772],["ET",352009],["IQ",30500446],["HT",753005],["NR",4765],["FM",19968],["PR",6209777],["MV",463279],["PW",6179],["LY",2185358],["NI",1022263],["MA",12393779],["RW",146797],["MR",425679],["PE",7439531],["MK",1459282],["MP",17871],["NP",5014232],["MD",848322],["QA",6062470],["ML",266277],["PS",1126577],["LI",44676],["NE",355102],["MZ",514765],["MU",836115],["PH",24537713],["MG",264777],["NO",2507874],["NL",9772073],["MC",45639],["RE",923422],["MT",543639],["PA",2941537],["LT",1144625],["NG",4127310],["MM",2946898],["YT",92756],["PL",7228613],["MW",61495],["OM",4680309],["NC",400648],["MN",870102],["RO",8092668],["MH",25736],["PG",87999],["LU",379265],["NU",3209],["NA",262543],["MX",68381785],["PT",4622234],["MY",22047572],["PK",12990338],["LR",86336],["NZ",4422917],["MS",7837],["RU",29212062],["MQ",540417],["PY",1457933],["MO",599820],["NF",1],["VC",128788],["TM",401475],["TJ",374779],["ME",739688],["ES",32008034],["VI",256059],["SN",888075],["AE",17636580],["TO",4471],["SE",6972355],["JE",173843],["SI",1029767],["VU",29403],["WS",9285],["TC",54113],["TZ",1086582],["A1",57501],["LK",1795155],["WF",6407],["SC",46925],["GB",80956251],["KN",106630],["TT",1764900],["CH",3720261],["GG",114658],["SB",6209],["VE",7801039],["SM",18159],["TV",18],["TH",38750701],["A2",71086],["SD",1911690],["YE",1653544],["SL",50574],["US",432563442],["LC",150375],["TN",2490465],["SY",2744358],["IM",129446],["SO",557101],["VN",18272323],["ST",32838],["UG",243196],["TL",20522],["AP",93692],["SR",442591],["ZM",213159],["SG",5876512],["UY",3292985],["PM",5910],["TR",28352405],["TW",9377874],["RS",3667823],["ZA",7777126],["VG",74063],["SA",54534442],["UA",9416636],["TG",86058],["EU",272409],["SZ",27416],["ZW",427772],["SK",2020592],["UZ",1545815]]'></google-chart>
        </template>
        <template is="dom-if" if="{{showTable}}" restamp>
            <h1>Stamping Data Table</h1>
            <h2>Data Table</h2>
            <google-chart type="table" data='[["Month","Days"],["Jan", 31],["Feb", 28],["Mar", 31],["Apr", 30],["May", 31],["Jun", 30]]'></google-chart>
        </template>
        <template is="dom-if" if="{{showBoth}}" restamp>
            <h1>Stamping Both</h1>
            <h2>Geo Chart</h2>
            <google-chart type="geo" data='[["Country","# of cows"],["BS",752348],["CD",306523],["AO",806287],["KY",169022],["IO",982],["BE",7240846],["CU",8951],["AW",237738],["CC",1],["AL",3170943],["KH",5039318],["BO",3677044],["BH",2528854],["CK",6496],["AI",20973],["CF",5877],["BN",1261279],["BZ",396333],["CY",1003070],["AU",18601713],["CO",25019570],["DZ",6032205],["CM",318071],["BA",1746715],["BD",6862277],["CR",4356245],["AG",248264],["TD",21008],["BG",3331495],["BJ",203453],["CZ",2748968],["AT",4009955],["KM",12892],["AS",31794],["CA",47546018],["BW",98467],["BB",613601],["CI",758435],["AR",23899962],["CL",17863532],["UN",4712275],["BF",166681],["BM",180303],["DK",3527352],["AZ",5370918],["CG",92149],["AD",96320],["CV",71066],["BR",65542588],["BY",2625345],["HR",3419193],["AM",915960],["CN",20931921],["AF",662182],["BI",30621],["BT",76527],["DJ",94734],["GR",4253871],["LA",3733866],["FR",27547672],["JP",38761999],["SV",2267621],["IS",321892],["GT",3862248],["GE",1306661],["KP",239],["FK",1181],["IE",6520811],["DM",113432],["VA",364],["GL",49078],["LV",781081],["GF",218059],["JO",4101559],["GQ",43056],["IN",37044982],["GN",543681],["DE",28552536],["KR",18184529],["FO",43597],["IL",8858476],["DO",6682727],["HN",2079527],["GD",123815],["LB",7304374],["PF",471257],["KZ",5851021],["ER",264],["ID",62250775],["GW",8619],["GH",1703842],["KW",6862643],["FJ",172913],["IT",23010367],["EC",6191501],["HK",10992110],["GP",509681],["LS",13607],["GA",230932],["KE",923662],["EE",1387922],["IR",14869013],["GY",298209],["GI",82716],["KG",946043],["FI",1952908],["JM",2698134],["EG",8721635],["HU",3543622],["GU",489182],["GM",66002],["KI",772],["ET",352009],["IQ",30500446],["HT",753005],["NR",4765],["FM",19968],["PR",6209777],["MV",463279],["PW",6179],["LY",2185358],["NI",1022263],["MA",12393779],["RW",146797],["MR",425679],["PE",7439531],["MK",1459282],["MP",17871],["NP",5014232],["MD",848322],["QA",6062470],["ML",266277],["PS",1126577],["LI",44676],["NE",355102],["MZ",514765],["MU",836115],["PH",24537713],["MG",264777],["NO",2507874],["NL",9772073],["MC",45639],["RE",923422],["MT",543639],["PA",2941537],["LT",1144625],["NG",4127310],["MM",2946898],["YT",92756],["PL",7228613],["MW",61495],["OM",4680309],["NC",400648],["MN",870102],["RO",8092668],["MH",25736],["PG",87999],["LU",379265],["NU",3209],["NA",262543],["MX",68381785],["PT",4622234],["MY",22047572],["PK",12990338],["LR",86336],["NZ",4422917],["MS",7837],["RU",29212062],["MQ",540417],["PY",1457933],["MO",599820],["NF",1],["VC",128788],["TM",401475],["TJ",374779],["ME",739688],["ES",32008034],["VI",256059],["SN",888075],["AE",17636580],["TO",4471],["SE",6972355],["JE",173843],["SI",1029767],["VU",29403],["WS",9285],["TC",54113],["TZ",1086582],["A1",57501],["LK",1795155],["WF",6407],["SC",46925],["GB",80956251],["KN",106630],["TT",1764900],["CH",3720261],["GG",114658],["SB",6209],["VE",7801039],["SM",18159],["TV",18],["TH",38750701],["A2",71086],["SD",1911690],["YE",1653544],["SL",50574],["US",432563442],["LC",150375],["TN",2490465],["SY",2744358],["IM",129446],["SO",557101],["VN",18272323],["ST",32838],["UG",243196],["TL",20522],["AP",93692],["SR",442591],["ZM",213159],["SG",5876512],["UY",3292985],["PM",5910],["TR",28352405],["TW",9377874],["RS",3667823],["ZA",7777126],["VG",74063],["SA",54534442],["UA",9416636],["TG",86058],["EU",272409],["SZ",27416],["ZW",427772],["SK",2020592],["UZ",1545815]]'></google-chart>
            <h2>Data Table</h2>
            <google-chart type="table" data='[["Month","Days"],["Jan", 31],["Feb", 28],["Mar", 31],["Apr", 30],["May", 31],["Jun", 30]]'></google-chart>
        </template>
    </template>
    <script>
        Polymer({
            is:'the-el',
            properties:{
                showTable:{
                    type:Boolean,
                    value:false
                },
                showGeo:{
                    type:Boolean,
                    value:false
                },
                showBoth:{
                    type:Boolean,
                    value:false
                }
            },
            _tTable:function(){
                this._t("showTable");
            },
            _tGeo:function(){
                this._t("showGeo");
            },
            _tBoth:function(){
                this._t("showBoth");
            },
            _t:function(p){
                this.set(p,!this.get(p));
            }
        })
    </script>
</dom-module>
```

*Repro*
1. Tap Stamp Geo
2. Tap Stamp Table

This makes the element very hard to re-use!

Let me know if you need anything else!

Thanks
@jongeho1

**Comment by ebidel** [comment]
Hmm, you're using `restamp`, so entirely new DOM should be created each time.
 Is this just for geo -> table or does it happen with all swaps of `type`?

**Comment by jongeho1** [comment]
@ebidel ```restamp``` was merely for exaggeration but can obviously happen when dealing w/ mutually exclusive conditions (across ```dom-if```) that are true at different times. The api on the window seems to be the commonality which is more so package dependent. It appears that loading packages in ```google-chart``` is expressed to be additive per instance but the behavior observed suggests it is more so immutable once loaded for the first time. With that said, I believe the package order could be problematic when stamped outside of the same iteration of the event loop:

corechart => table,gauge,treemap

Feel free to change "geo" to "pie" and others and see a similar behavior.

FYI, this is not hard to workaround but definitely makes this element hard to re-use w/ this limitation since ```table``` and ```corechart``` items are used together often.

Let me know if you have any questions!

@jongeho1

**Comment by ebidel** [comment]
Repo using 1.2.1: http://jsbin.com/payopoquvu/edit?html,output

**Comment by steven-ferguson** [comment]
This looks like an [existing issue](https://github.com/google/google-visualization-issues/issues/1755) with trying to make multiple calls to ```google.load```. I think loading all the required packages will resolve this issue.

**Comment by wesalvaro** [comment]
I believe this was fixed with #83.
I was able to reproduce it before but not after.
Let me know if you are still experiencing the issue.
